### PR TITLE
Fix GodMode not being displayed when Vehicle-GodMode

### PR DESCRIPTION
**Pull request by pierrelasse**
Fixes https://github.com/YimMenu/YimMenu/issues/3117 (hopefully)

*Let's have github actions build it for me and then test it :D*

### CODE REVIEW

**Comment by pierrelasse** [comment]
Plz approve the workflow if possible :))

**Comment by gir489returns** [comment]
HAHAHAHAHAHAHA

**Review by Yimura — APPROVED** [review]

**Review by rkwapisz — COMMENTED** [review]
My hint in https://github.com/YimMenu/YimMenu/issues/3117#issuecomment-2111690283 wasn't intended to be the complete solution.

Consider what the user will see given how mode_str is appended and how the translation keys are formatted.

In this case, a user will see "GODVehicle God Mode" because there is no additional white space being placed in front of VEHICLE_GOD if player god mode is also enabled.

**Comment by rkwapisz** [comment]
> Fixes #3117 (hopefully)
> 
> _Let's have github actions build it for me and then test it :D_

Please please please set up your own dev environment so you can compile and test on your own machine.
Even such a trivial change like this should be tested, because then you'd have been able to see why it warrants a little more adjustment.

**Review by Yimura — CHANGES_REQUESTED** [review]
Retracting my approval, we need to first test how it looks in-game.

**Review by rkwapisz — COMMENTED** [review]
```
if (g.esp_player.god && (ped_damage_bits & (uint32_t)eEntityProofs::GOD)) // If player is in god mode (meaning GOD has been appended to mode_str 
{
	mode_str += " | "; // Add some kind of text separator
}

mode_str += "ESP_VEH_GOD"_T.data(); // Append vehicle god mode text
```

**Comment by pierrelasse** [comment]
Translation changes: https://github.com/YimMenu/Translations/pull/159

**Comment by pierrelasse** [comment]
Without anything
image

Vehicle Godmode
image

Godmode and Vehicle Godmode
image

Bullet & Explosion proof
image

File: src/views/esp/view_esp.cpp
Comment (Yimura) [inline src/views/esp/view_esp.cpp]: Doesn't match the coding guidelines, please switch this to `snake_case`.

**Review by Yimura — CHANGES_REQUESTED** [review]

**Comment by pierrelasse** [comment]
The requested changes were made.

File: src/views/esp/view_esp.cpp
Comment (Yimura) [inline src/views/esp/view_esp.cpp]: Imagine compiling before committing bruh, what a waste of my time....

**Review by Yimura — CHANGES_REQUESTED** [review]

File: src/views/esp/view_esp.cpp
Comment (pierrelasse) [reply src/views/esp/view_esp.cpp]: Oh cool. Why didnt i compile 🤔 Sry

Comment (Yimura) [reply src/views/esp/view_esp.cpp]: 1. I requested the change to snake case
2. You change it only in 1 spot
3. Commit without compiling locally
4. Build fails because you didn't change it to snake case everywhere

Comment (gir489returns) [reply src/views/esp/view_esp.cpp:88]: @Yimura You can always tell the extra shitty commits when:

>This commit was created on GitHub.com and signed with GitHub’s verified signature.

Comment (pierrelasse) [reply src/views/esp/view_esp.cpp]: > 1. I requested the change to snake case
> 2. You change it only in 1 spot
> 3. Commit without compiling locally
> 4. Build fails because you didn't change it to snake case everywhere

I know. I will change it when im on pc. If vs would be less trash, it would compile fastet than in a few minutes and linking without it taking a few minutes and while all thst taking 90% cpu. Maybe xou understand that i just edited the file without compiling it for 10mins.

**Comment by pierrelasse** [comment]
image

**Review by Yimura — APPROVED** [review]

**Comment by Yimura** [comment]
> Maybe xou understand that i just edited the file without compiling it for 10mins.

So you lose less time by committing immediately, then CI fails to build it (which first requires approval from a collaborator). Then you lose more time by having to go in and make changes to fix the issue losing you more time.

Make it make sense, sometimes it makes sense to lose some time up front to lose less in the future. No wonder people stop contributing because they try to take short cuts...

Don't forget the part where @YimMenu/reviewers need to go through these things and look at what you're making is sound, if you make a mistake in a review you will lose time. If you keep doing this I and other reviewers will put less effort in reviewing the PR (as you may have noticed from the time between me reviewing this each time).

**Comment by lonelybud** [comment]
<del>LGTM</del>

the translations of `VEH` is concerning - https://github.com/YimMenu/Translations/pull/159 & blocking

**Review by Yimura — APPROVED** [review]

**Pull request merged**
